### DISCUSSION TITLE
Support IPMI power commands

### DISCUSSION THREAD

**Discussion by stv0g** [seriousm4x/UpSnap#452] [Feature requests]
I would be great if we can extend UpSnap with the capability to send IPMI commands to control the power state of remote systems.

Most servers nowadays can be remotely managed by a Board Management Controller (ILO, iDRAC, ...). Most of them support the Intelligent Platform Management Interface (IPMI).

- There exist some quite good Go packages for IPMI. E.g.
  - https://github.com/vmware/goipmi
  - https://github.com/gebn/bmc

Thanks to a BMC we could implement even more cool things like:

- Auto-discovery of hosts
- Detecting power state of hosts

**Comment by seriousm4x** [comment]
I haven't heard of IPMI yet, but I think your suggestion is currently possible.

About the two linked repos:
- The first one no longer looks maintained. At least it hasn't had a commit for 6 (or 9) years. I wouldn't like to implement something like that, even if the standard hasn't changed since then.
- The same applies to the second repo. The last commit is up to date, but the repo has been archived. I therefore do not expect it to receive any further updates.

What I would suggest: Have a look at the readme [here](https://github.com/seriousm4x/UpSnap?tab=readme-ov-file#install-additional-packages-for-shutdown-cmd). There is an [Alpine package called "ipmitool"](https://pkgs.alpinelinux.org/packages?name=ipmitool&branch=v3.19&repo=&arch=&maintainer=). You can install that. Every device in UpSnap has the possibility to set a custom command for ping, wake and shutdown. You could enter your ipmi command there.

**Reply by stv0g** [reply]
Thanks, I will try it with ipmitool. How about including it into your Docker image by default?

**Comment by stv0g** [comment]
After #453 has not been merged, I added the missing packages by building my own Docker image:

```Dockerfile
FROM ghcr.io/seriousm4x/upsnap:4

RUN apk add ipmitool
```
